src/cpufreq.c

```diff
@@ -28,18 +31,52 @@
 
 static int num_cpu = 0;
 
+static char const * freq_fname = "/sys/devices/system/cpu/cpu%d/cpufreq/"
+								 "scaling_cur_freq";
+static char const * freq_fname_rhel = "/sys/devices/system/cpu/cpu%d/cpufreq/"
+									  "cpuinfo_cur_freq";
+
+static int setup_freq_fname()
+{
+	int status;
+	char filename[256];
+
+	status = ssnprintf (filename, sizeof (filename),
+						freq_fname, 0);
+	if ((status < 1) || ((unsigned int)status >= sizeof (filename)))
+		return 0;
+
+	if (!access (filename, R_OK))
+		return 1;
+
+	status = ssnprintf (filename, sizeof (filename),
+						freq_fname_rhel, 0);
+	if ((status < 1) || ((unsigned int)status >= sizeof (filename)))
+		return 0;
+
+	if (!access (filename, R_OK)) {
+		freq_fname = freq_fname_rhel;
+		return 1;
+	}
+
+	return 0;
+}
+
 static int cpufreq_init (void)
 {
-        int status;
+	int status;
 	char filename[256];
 
+	if ( !setup_freq_fname() ) {
```

Comment:
Nit: Please don't use spaces after ( and before ) (i.e. no "padding" inside the parenthesis).